Removed duplicated blocks of code

gonzaloUr opened the pull request:
Moved duplicated hashCode method in MacroBlock and MacroMakerBlock to a common class.

tmortagne commented on xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/AbstractMacroBlock.java:
```diff
@@ -0,0 +1,43 @@
+ */
+public abstract class AbstractMacroBlock extends AbstractBlock
+{
+
```
I doubt this is working since that would mean MacroBlock and MacroMakerBlock are calling a super constructor that does not exist.

tmortagne commented on xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/AbstractMacroBlock.java:
```diff
@@ -0,0 +1,43 @@
+ * Common class to MacroBlock and MacroMakerBlock.
+ *
+ * @version $Id$
+ * @since 9.11
```
9.11 being out it now should be `10.0RC1`

tmortagne commented on xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/AbstractMacroBlock.java:
```diff
@@ -0,0 +1,43 @@
+        HashCodeBuilder builder = new HashCodeBuilder();
+
+        builder.appendSuper(super.hashCode());
+        builder.append(getContent());
```
You are calling methods which don't exist at this level.

tmortagne commented:
AbstractMacroBlock is the right move but you are missing a lot of stuff in that class. Please don't commit anything until the build passes with Maven.

gonzaloUr commented:
I've already changed what was suggested and also added a unit test. My changes seem to build (including the test) but I'm not able to get the entire project. I've attached the full log as well as the parts where the unit test passes, because of this I'm pretty sure that the errors have nothing to do with my changes.

[log.txt](https://github.com/xwiki/xwiki-rendering/files/1581341/log.txt)
[log-test.txt](https://github.com/xwiki/xwiki-rendering/files/1581342/log-test.txt)

tmortagne commented on xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/AbstractMacroBlock.java:
```diff
@@ -0,0 +1,150 @@
+    /**
+     * @return the macro content.
+     */
+    public abstract String getContent();
```
It does not make much sense to have all those abstract method. You should move the field since they are duplication too.

tmortagne commented on xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/AbstractMacroBlock.java:
```diff
@@ -0,0 +1,150 @@
+     * @param childrenBlocks the list of children blocks of the block to construct
+     * @param parameters the parameters to set
+     */
+    public AbstractMacroBlock(List<? extends Block> childrenBlocks, Map<String, String> parameters)
```
You don't need all those constructors. MacroBlock and MacroMarkerBlock only need one.

gonzaloUr commented:
I removed the constructors and abstract methods and moved the fields and getters to AbstractMacroBlock.

tmortagne commented on xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/AbstractMacroBlock.java:
```diff
@@ -0,0 +1,126 @@
+    /**
+     * @return the macro content.
+     */
+    public String getContent() {
```
This does not follow XWiki Java codestyle (you are missing a newline, see the classes from which you copy/pasted that). See http://dev.xwiki.org/xwiki/bin/view/Community/JavaCodeStyle#HConfiguringyourIDEtousetheXWikicodestyle for resource to setup Eclipse or IDEA formatters.

tmortagne commented on xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/MacroBlock.java:
```diff
@@ -128,29 +83,16 @@ public boolean equals(Object obj)
     }
 
     @Override
     public int hashCode()
```
Why not implementing hashCode at abstract level since there is nothing specific here.

gonzaloUr commented:
I had to override equals in AbstractMacroBlock and hashCode in both MacroBlock and MacroMakerBlock. It is not allowed by the checkstyle for a class to override equals but not hashCode and vice versa.

tmortagne commented:
Thanks for your contribution !
